diff --git a/external/qt/qt_static_plugins/kimageformats/CMakeLists.txt b/external/qt/qt_static_plugins/kimageformats/CMakeLists.txt
index 81d58bd..add7909 100644
--- a/external/qt/qt_static_plugins/kimageformats/CMakeLists.txt
+++ b/external/qt/qt_static_plugins/kimageformats/CMakeLists.txt
@@ -60,6 +60,7 @@ if (DESKTOP_APP_USE_PACKAGED OR LINUX)
         $<TARGET_NAME_IF_EXISTS:avif>
         $<TARGET_NAME_IF_EXISTS:PkgConfig::DESKTOP_APP_AVIF>
         $<TARGET_NAME_IF_EXISTS:heif>
+        desktop-app::external_ffmpeg
         $<TARGET_NAME_IF_EXISTS:PkgConfig::DESKTOP_APP_JXL>
     )
 elseif (WIN32)
