diff --git a/codegen/lang/generator.cpp b/codegen/lang/generator.cpp
index 07e5b60..f22452d 100644
--- a/codegen/lang/generator.cpp
+++ b/codegen/lang/generator.cpp
@@ -185,7 +185,7 @@ inline constexpr auto kKeysCount = ushort(" << keysCount_ << ");\n\
 
 bool Generator::writeAllKeys() {
 	auto file = common::CppFile(basePath_ + "_keys.h", project_);
-	file.include(baseName_ + ".h").newline();
+	file.includeFromLibrary("array").include(baseName_ + ".h").newline();
 	file.pushNamespace("tr").newline();
 	for (const auto &[combination, code] : specializations_) {
 		file.stream() << code;
@@ -193,6 +193,43 @@ bool Generator::writeAllKeys() {
 	for (const auto &declaration : declarations_) {
 		file.stream() << declaration << "\n";
 	}
+	auto brandedIndices = QStringList();
+	for (auto i = 0, count = int(langpack_.entries.size()); i != count; ++i) {
+		const auto &entry = langpack_.entries[i];
+		const auto goodKey = [&](const QString &value) {
+			return entry.key.contains(value);
+		};
+		if (entry.value.contains("Telegram")
+			&& (goodKey("_tray")
+				|| goodKey("lng_tray_icon_text")
+				|| goodKey("lng_error_start_minimized_passcoded")
+				|| goodKey("lng_proxy_unsupported")
+				|| goodKey("lng_bad_photo")
+				|| goodKey("lng_update_telegram")
+				|| goodKey("lng_sure_save_language")
+				|| goodKey("lng_settings_auto_start")
+				|| goodKey("lng_settings_add_sendto")
+				|| goodKey("lng_theme_no_desktop")
+				|| goodKey("lng_download_path_default_radio")
+				|| goodKey("lng_passcode_about")
+				|| goodKey("lng_proxy_sponsor_about")
+				|| goodKey("lng_message_unsupported")
+				|| goodKey("lng_bot_share_location_unavailable")
+				|| goodKey("lng_new_version_wrap")
+				|| goodKey("lng_theme_editor_need_unlock")
+				|| goodKey("lng_payments_not_supported")
+				|| goodKey("lng_group_call_mac_access")
+				|| goodKey("lng_language_not_ready_about")
+				|| goodKey("lng_outdated_")
+				|| goodKey("lng_mac_menu_hide_"))) {
+			brandedIndices.push_back(QString::number(indices_[i]));
+		}
+	}
+	file.stream() << "\ninline constexpr std::array<int, "
+		<< brandedIndices.size()
+		<< "> hasTelegram = { "
+		<< brandedIndices.join(", ")
+		<< " };\n";
 	file.newline().popNamespace();
 	return file.finalize();
 }
